runtime/runtime/src/verifier.rs

```diff
@@ -105,8 +104,11 @@ pub fn verify_and_charge_transaction(
         }
     }
 
-    if let Err(amount) = check_rent(&signer_id, &signer, &config, apply_state.epoch_length) {
-        return Err(InvalidTxError::RentUnpaid { signer_id: signer_id.clone(), amount }.into());
+    if let Err(amount) = check_storage_cost(&signer_id, &signer, &config, apply_state.epoch_length)
```

Comment:
May be I don't understand the exact flow, but this seems to be checked before the transaction is executed. Won't it discard a transaction from an account that doesn't have NOW enough balance for storage, but will have as a result of applying the transaction? In particular, won't this condition rule out transactions that are intentionally sent to add funds to have enough for storage?

A test for that would also be great :)

Reply:
Yes, we need to check it at the end of the transaction/receipt not at the beginning. CC @evgenykuzyakov

Reply:
Your own transaction can't increase your funds. No matter how hard you're trying :)

Reply:
@skidanovalex @nearmax This is verification of transaction validity. Transaction is invalid only in the case if signer account doesn't have enough funds. In cases when you add funds to under-funded account - signer will be account with funds (as @evgenykuzyakov points out "your own transaction can't increase your funds").